.ddev/commands/cypress/cypress-run

```diff
@@ -2,6 +2,19 @@
 ## Description: Use Cypress in "runner" mode.
 ##
 ## Usage: cypress-run [arguments]
-## Example: "ddev cypress-run --browser chrome"
+## Example: "ddev cypress-run"
+## This executes only the functional test cases using Electron.
 
-DISPLAY= cypress run "$@"
+# If this is run from the command line, these variables may not be set.
+# Set them to defaults. It may be required to reset the user password
+# using drush, like so:
+# `ddev drush upwd ct-admin ct-admin`
+# Or simply set the actual password in the environment variable.
+if [ -z "${CYPRESS_ADMIN_USERNAME}" ]; then
+  CYPRESS_ADMIN_USERNAME=ct-admin
+fi
+if [ -z "${CYPRESS_ADMIN_PASSWORD}" ]; then
+  CYPRESS_ADMIN_PASSWORD=ct-admin
+fi
+
+DISPLAY= CYPRESS_ADMIN_USERNAME=${CYPRESS_ADMIN_USERNAME} CYPRESS_ADMIN_PASSWORD=${CYPRESS_ADMIN_PASSWORD} cypress run --spec "cypress/e2e/*.cy.js" --browser electron "$@"
```

Comment:
@zeshanziya @KosalaiV, `chrome` didn't work for me locally. I had to switch to `electron`. How did you get `chrome` working?

Reply:
> @zeshanziya @KosalaiV, `chrome` didn't work for me locally. I had to switch to `electron`. How did you get `chrome` working?

`DISPLAY= CYPRESS_ADMIN_USERNAME=${CYPRESS_ADMIN_USERNAME} CYPRESS_ADMIN_PASSWORD=${CYPRESS_ADMIN_PASSWORD} cypress run --spec "cypress/e2e/*.cy.js" --browser chrome "$@"` should work

Reply:
@hussainweb I haven't tried Chrome yet. I'm also concerned about hardcoding `--spec "cypress/e2e/*.cy.js"` in the command file. If we plan to run a different set of tests, we'll need to create another command. It would be better to keep it generic, as we can pass any extra arguments to the command as needed.

Reply:
@KosalaiV, it says it couldn't find chrome on my system. It makes sense too because only `electron` is available inside the container.

Reply:
@zeshanziya (and @KosalaiV),

I think we can skip `--spec` entirely. Doesn't `cypress.config.js` set that path as the default anyway?

Reply:
I tested locally and it can pick up the specs properly without the argument. Let's just leave it out.

Reply:
@hussainweb I remember, Initially I tried chrome but it didn't work. I am not sure but I read somewhere that Chrome cannot be installed on a container when the host is Mac. I will try to find the link and share it.

Reply:
@zeshanziya, no need to worry about this. Electron works just fine, no? @KosalaiV, any concerns with Electron?

Reply:
@hussainweb @zeshanziya It was working, Please check this [run](https://github.com/contrib-tracker/backend/actions/runs/10197171049/job/28209453856). Might get some issue executing in local. But works in pipeline.
![Screen Shot 2024-08-05 at 12 58 36 PM](https://github.com/user-attachments/assets/fab55c6e-cc2b-4519-8542-b60db36fbce2)

Reply:
@KosalaiV, yes, that's the weird part. Can you get it working locally? That's also important.

Reply:
@hussainweb Yes, works locally. It requires Chrome to be installed on your system.

Reply:
@hussainweb So It has an image `cypress/included:cypress-13.12.0-node-20.14.0-chrome-126.0.6478.114-1-ff-127.0.1-edge-126.0.2592.61-1` with all browsers installed. I tried using it, but when it didn't work, I didn't investigate further since Electron was working fine.

Reply:
@KosalaiV, Chrome is installed on my system. It didn't work for me. And like I said, it makes sense because Chrome is not in the container. Please test by running it through DDEV, not directly.

Reply:
@KosalaiV, what's the problem with Electron?

Reply:
> @KosalaiV, what's the problem with Electron?

@hussainweb @zeshanziya As Electron is not a real browser that end-users would use, so it might not catch all issues that could appear in a real browser environment.

Reply:
> @hussainweb Yes, works locally. It requires Chrome to be installed on your system.

@KosalaiV  Are you running it using `ddev`? I believe it will work without ddev. So If we just use `npx cypress` or `npm run` to run the cypress, it will install the browser on local host machine if not found. Some issue when running inside container.